numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -3093,6 +3093,12 @@ VOID_compare(char *ip1, char *ip2, PyArrayObject *ap)
     if (!PyArray_HASFIELDS(ap)) {
         return STRING_compare(ip1, ip2, ap);
     }
+    const PyDataMem_Handler *mem_handler = PyDataMem_GetHandler(NULL);
+    if (mem_handler == NULL) {
+        /* Better fallback to default than goto finish. */
+        PyErr_Clear();
+        mem_handler = &default_handler;
+    }
```

Comment:
`VOID_compare` returns `int`. On error, the function finishes prematurely. Should this happen here too (returning res=0)? I didn't found any reference to it, where it's being used?

Reply:
Yes, I think returning early with an error set is the right thing to do.

Reply:
You can't find a reference to it because `VOID_compare` is likely used in templated code as something like `@type@_compare` or `VOID_@op@`.